components/CreateClassModal/CreateClassModal.js

```diff
@@ -0,0 +1,169 @@
+import { useState } from 'react';
+import {
+  FormControl,
+  InputLabel,
+  MenuItem,
+  TextField,
+  Select,
+  DialogTitle,
+  Dialog,
+  DialogContent,
+  Stack,
+  } from '@mui/material';
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import DateTimePicker from '@mui/lab/DateTimePicker';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import MainButton from '../basecomponents/MainButton.js'
+import styles from '../../utils/styles/CreateClassModalStyles/CreateClassModalStyles.module.css';
+
+
+export default function CreateClassModal() {
+  const [subject, setSubject] = useState('');
+  const [type, setType] = useState('');
+  const [newStartTime, setNewStartTime] = useState(null);
+  const [newEndTime, setNewEndTime] = useState(null);
+  const [classObj, setClassObj] = useState({});
+  const [open, setOpen] = useState(false);
+
+  const handleChange = (e) => {
+    if(e.target.name === 'capacity') {
+      setClassObj({ ...classObj, [e.target.name]: Number(e.target.value) })
+    } else {
+      setClassObj({ ...classObj, [e.target.name]: e.target.value })
```

Comment:
Tech debt: I think this would be a perfect case for useReducer hook. The function you use here is exactly the logic that a reducer dispatch function does, instead of an if else statement, it uses a switch case. Not needed now but make a trello ticket to refactor this later to useReducer

Reply:
There are only 2 cases, whether it is 'capacity' or not, useReducer would be a lot of extra lines of code and refactoring across all form fields. We are okay to implement it but I think this way is cleaner since the only exception is when e.target.name is 'capacity' and it needs to be a number